src/chart-addons/stacked-legend.js

```diff
@@ -82,26 +82,53 @@ function setColorScaleDomain_v1(domain) {
   }
 }
 
-export function getLegendStateFromChart(chart, useMap) {
+export function getLegendStateFromChart(chart, useMap, selectedLayer) {
+
```

Comment:
@biovisualize,  I was trying to get only relevant chart state by calling `const state = getLegendStateFromChart(_chart, useMap, selectedLayer)` only for current selected layer for the chart, but `_chart._doRender` is called from many different options. Thus, it is difficult to capture currentLayer from immerse and mapd-charting in all those cases. For this reason, only option to remove extra legends is removing in this funtion.